### PR TITLE
Wip modular backends

### PR DESCRIPTION
**Pull request by christian-marie**
Do not merge yet, still need to find a solution to the multiple projects in one repo problem.

### CODE REVIEW

File: lib/Git/Vogue/PluginDiscoverer/Libexec.hs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+libExecDiscoverer libexec_dir =
+    PluginDiscoverer (discover libexec_dir) disable enable
+
+-- | Find all plugins within the libexec dir.
+--
+-- This function inspects the $PREFIX/libexec/git-vogue directory and the
+-- directories listed in the $GIT_VOGUE_PATH environmental variable (if
+-- defined) and builds a 'Plugin' for the executables found.
+--
+-- Files that are set non executable or included in the
```
Comment (thsutton) [inline lib/Git/Vogue/PluginDiscoverer/Libexec.hs]: Please add ellipsis or some more words.

File: lib/Git/Vogue/PluginDiscoverer/Libexec.hs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+runPlugin :: MonadIO m => FilePath -> String -> [FilePath] -> m Result
+runPlugin plugin cmd paths = liftIO $ do
+    (status, out, err) <- readProcessWithExitCode plugin [cmd] (unlines paths)
+    let glommed = fromString $ out <> err
+
+    return $ case status of
+        ExitSuccess   -> Success glommed
+        ExitFailure 1 -> Failure glommed
+        ExitFailure n -> Catastrophe n glommed
+
```
Comment (thsutton) [inline lib/Git/Vogue/PluginDiscoverer/Libexec.hs]: So much space!

File: lib/Git/Vogue/Types.hs [hunk 3]
```diff
@@ -22,18 +50,31 @@ data Check
...
+instance Show (Plugin m) where
+    show Plugin{..} =
+        T.unpack (unPluginName pluginName)
+            <> if enabled then mempty else " (disabled)"
+
+instance Eq (Plugin m) where
+    a == b = pluginName a == pluginName b
+
+instance Ord (Plugin m) where
+    compare a b = compare (pluginName a) (pluginName b)
```
Comment (thsutton) [inline lib/Git/Vogue/Types.hs]: compare `on` pluginName?

File: lib/Git/Vogue/Types.hs [hunk 4]
```diff
@@ -22,18 +50,31 @@ data Check
...
+    , runFix     :: [FilePath] -> m Result
+    }
+
+instance Show (Plugin m) where
+    show Plugin{..} =
+        T.unpack (unPluginName pluginName)
+            <> if enabled then mempty else " (disabled)"
+
+instance Eq (Plugin m) where
+    a == b = pluginName a == pluginName b
```
Comment (thsutton) [inline lib/Git/Vogue/Types.hs]: (==) `on` pluginName?

**Comment by christian-marie** [comment]
Comments addressed in #63

**Pull request merged**